app/views/NewExploreDeep/index.tsx

```diff
@@ -138,234 +208,595 @@ const representationOptions: Option[] = [
+                                Download
+                            </ButtonLikeLink>
+                        </div>
+                    ));
+                } else if (response?.genericExport?.status === 'FAILURE') {
+                    removeAlert(DOWNLOAD_ALERT_NAME);
+                    setExportIdToDownload(undefined);
+                    alert.show(
+                        'There was an issue creating the export.',
+                        { variant: 'error' },
+                    );
+                }
+            },
+        },
+    );
+
+    const [
+        createExport,
+        { loading: exportCreatePending },
+    ] = useMutation<GenericExportCreateMutation, GenericExportCreateMutationVariables>(
+        GENERIC_EXPORT_CREATE,
+        {
+            onCompleted: (response) => {
+                if (!response?.genericExportCreate || !response?.genericExportCreate?.ok) {
+                    alert.show(
+                        'There was an issue creating the export.',
+                        { variant: 'error' },
+                    );
+                    setExportIdToDownload(undefined);
+                }
+                if (
+                    response?.genericExportCreate?.ok
+                    && response?.genericExportCreate?.result?.id
+                ) {
+                    setExportIdToDownload(response.genericExportCreate.result.id);
+                    addAlert({
+                        variant: 'info',
+                        duration: Infinity,
+                        name: DOWNLOAD_ALERT_NAME,
+                        children: 'Please wait while the export is being prepared.',
+                    });
+                }
+            },
+            onError: () => {
+                alert.show(
+                    'There was an issue creating the export.',
+                    { variant: 'error' },
+                );
+                setExportIdToDownload(undefined);
+            },
+        },
+    );
+
+    useEffect(
+        () => {
+            const shouldPoll = exportIdToDownload
+                && genericExportData?.genericExport?.status !== 'SUCCESS'
+                && genericExportData?.genericExport?.status !== 'FAILURE';
+
+            if (shouldPoll) {
+                startPolling(5000);
+            } else {
+                stopPolling();
+            }
+        },
+        [
+            removeAlert,
+            exportIdToDownload,
+            genericExportData,
+            startPolling,
+            stopPolling,
+        ],
+    );
+
     const handleImageExportClick = useCallback(() => {
-        console.warn('jpeg export clicked');
-    }, []);
+        // FIXME: Implement JPEG Export
+        showPrintPreview();
+    }, [
+        showPrintPreview,
+    ]);
 
     const handlePdfExportClick = useCallback(() => {
-        console.warn('pdf export clicked');
-    }, []);
+        showPrintPreview();
+    }, [
+        showPrintPreview,
+    ]);
```

Comment:
Any reason to create new functions?

Reply:
its for future

Reply:
Is this a work in progress? Are we modifying the function before the release?

Reply:
We need to modify these before release.

Reply:
Waiting for the modifications.